moped-editor/src/views/projects/projectView/ProjectPhases.js

```diff
@@ -216,90 +241,103 @@ const ProjectPhases = ({
         },
       }}
       editable={{
-        onRowAdd: (newData) => {
-          const newPhaseObject = Object.assign(
-            {
-              project_id: projectId,
-              completion_percentage: 0,
-              completed: false,
-            },
-            newData
-          );
+        onRowAdd: async (newData) => {
+          setIsMutating(true);
+          const { moped_phase, ...rest } = newData;
 
-          // if necessary, updates existing phases in table to ensure only one is marked "current"
-          updateExistingPhases(newPhaseObject);
+          const newPhasePayload = {
+            project_id: projectId,
+            completion_percentage: 0,
+            completed: false,
+            phase_id: moped_phase.phase_id,
+            ...rest,
+          };
 
+          // if necessary, updates existing phases in table to ensure only one is marked "current"
+          if (newPhasePayload.is_current_phase) {
+            const projPhasesIdsToUpdate = getCurrentPhaseIDs(
+              null,
+              data.moped_proj_phases
+            );
+            if (projPhasesIdsToUpdate.length > 0) {
+              await updateExistingPhases(projPhasesIdsToUpdate);
+            }
+          }
           // Execute insert mutation, returns promise
-          return addProjectPhase({
+          await addProjectPhase({
             variables: {
-              objects: [newPhaseObject],
+              objects: [newPhasePayload],
             },
-          }).then(() => {
-            // Refetch data
-            refetch();
-            !!projectViewRefetch && projectViewRefetch();
+          }).catch((err) => {
+            console.error(err);
           });
+          // Refetch data
+          await refetch();
+          if (!!projectViewRefetch) {
+            await projectViewRefetch();
+          }
+          setIsMutating(false);
         },
-        onRowUpdate: (newData, oldData) => {
-          const updatedPhaseObject = {
-            ...oldData,
-          };
-          // Array of differences between new and old data
-          let differences = Object.keys(oldData).filter(
-            (key) => oldData[key] !== newData[key]
-          );
-
-          // Loop through the differences and assign newData values.
-          // If one of the Date fields is blanked out, coerce empty
-          // string to null.
-          differences.forEach((diff) => {
-            let shouldCoerceEmptyStringToNull =
-              newData[diff] === "" &&
-              (diff === "phase_start" || diff === "phase_end");
```

Comment:
to summarize these changes:
- we always commit to the DB all of the newData, regardless of what (if any) column values have changed
- all empty strings are converted to `null` - no need to overthink that at a column level